presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -1540,6 +1542,16 @@ public SystemSessionProperties(
                         false,
                         value -> RandomizeOuterJoinNullKeyStrategy.valueOf(((String) value).toUpperCase()),
                         RandomizeOuterJoinNullKeyStrategy::name),
+                integerProperty(
+                        RANDOMIZE_OUTER_JOIN_NULL_KEY_NULL_COUNT_THRESHOLD,
+                        "Enable randomizing null join key for outer join when number of null join keys exceed the threshold",
+                        100_000,
+                        false),
+                doubleProperty(
```

Comment:
can we simplify this and only have one parameter to control? we can keep the ratio parameter and choose not to apply the transformation if the overall cardinality of the input is small

Reply:
I assume that we will need some tuning for these parameters. Having more flexibility could help here.

Reply:
I understand the flexibility argument, I'm just worried there are too many parameters creates an exponential number of combinations to tune.
What is the use case here that is not covered by just tuning the ratio?